data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -669,6 +760,202 @@ class ApiServerClient {
       })
       .then(resource => _.get(resource, 'spec.options.context'));
   }
+
+  /**
+   * @description Create OSB Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Name of resource group 
+   * @param {string} opts.resourceType - Type of resource 
+   * @param {string} opts.resourceId - Unique id of resource 
+   * @param {string} opts.metadata - Optional; pass finalizers or some other field
+   * @param {string} opts.labels - to be put in label
+   * @param {Object} opts.spec - Value to set for spec field of resource
+   * @param {string} opts.status - status of the resource
+   */
+  // Note:- In this method, keys in ServiceInstance CR are required to be camelcased
+  // Hence while creating resource, osb keys (snakecased) translated into camelcased using camelcase-keys module
+  createOSBResource(opts) {
+    logger.info(`Creating OSB resource with opts: `, opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to create resource`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to create resource`);
+    assert.ok(opts.resourceId, `Property 'resourceId' is required to create resource`);
+    assert.ok(opts.spec, `Property 'spec' is required to create resource`);
+    const metadata = _.merge(opts.metadata, {
+      name: opts.resourceId
+    });
+    if (opts.labels) {
+      metadata.labels = opts.labels;
+    }
+    const crdJson = this.getCrdJson(opts.resourceGroup, opts.resourceType);
+    const resourceBody = {
+      apiVersion: `${crdJson.spec.group}/${crdJson.spec.version}`,
+      kind: crdJson.spec.names.kind,
+      metadata: metadata,
+      spec: camelcaseKeys(opts.spec)
+    };
+
+    if (opts.status) {
+      _.forEach(opts.status, (val, key) => {
+        if (key === 'state') {
+          resourceBody.metadata.labels = _.merge(resourceBody.metadata.labels, {
+            'state': val
+          });
+        }
+      });
+      resourceBody.status = opts.status;
+    }
+    // Create Namespace if not default
+    const namespaceId = this.getNamespaceId(opts.resourceType === CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS ?
+      _.get(opts, 'spec.instance_id') : opts.resourceId
+    );
+    // Create Namespace if not default
+    return this.init()
+      .then(() => this.createNamespace(namespaceId))
+      .then(() => apiserver
+        .apis[opts.resourceGroup][CONST.APISERVER.API_VERSION]
+        .namespaces(namespaceId)[opts.resourceType].post({
+          body: resourceBody
+        }))
+      .catch(err => {
+        return convertToHttpErrorAndThrow(err);
+      });
+  }
+
+  /**
+   * @description Update OSB Resource in Apiserver with the opts
```

Comment:
There is a different patch methind. Does this mean that this function overwrites the values? If so, can you please specify in comment?

Reply:
I though update and patch/merge were intuitive enough. Will add a comment anyway.